src/delfino/main.py

```diff
@@ -83,17 +90,76 @@ def invoke(self, ctx: click.Context) -> Any:
     def get_help(self, *args, **kwargs) -> str:
         help_str = super().get_help(*args, **kwargs)
 
-        if self._hidden_plugins:
+        if self._hidden_plugin_commands:
             if logging.root.level == logging.DEBUG:
                 help_str += click.style(
-                    f"\n\nDisabled command{'s' if len(self._hidden_plugins) > 1 else ''}: "
-                    + ", ".join(sorted(self._hidden_plugins))
-                    + f" (see 'tool.{ENTRY_POINT}.disable_commands' in '{PYPROJECT_TOML_FILENAME}')",
+                    f"\n\nDisabled command{'s' if len(self._hidden_plugin_commands) > 1 else ''}: "
+                    + ", ".join(sorted([f"{plugin}.{cmd}" for plugin, cmd in self._hidden_plugin_commands.items()]))
+                    + f" (see 'tool.{ENTRY_POINT}.disable_plugin_commands' in '{PYPROJECT_TOML_FILENAME}')",
                     fg="white",
                 )
 
         return help_str
 
+    def _get_commands(self) -> Dict[str, click.Command]:
+        if self._commands:
+            return self._commands
+
+        command_packages = [
+            CommandPackage(package=self.PRESET_COMMAND_PACKAGE, plugin_name="delfino"),
+            *list(discover_packages()),
+            CommandPackage(package=COMMANDS_DIRECTORY_NAME, required=False),
+            CommandPackage(package="tasks", required=False, new_name=COMMANDS_DIRECTORY_NAME),
+        ]
+
+        registry = _CommandRegistry(self._hidden_plugin_commands)
+        for command_package in command_packages:
+            registry.register_command_package(command_package)
+
+        self._commands = registry.as_dict()
```

Comment:
A refactoring suggestion - no need to keep the original type that was a dict. If you need something more advanced than a dict, use it directly. E.g. just drop the dict and use the registry everywhere. It will simplify the code a lot. Then have a look at where and how the registry is called. You can simplify the code even further by moving implementation into the registry, if it is a concern of that class.